matpower/matpower-converter/src/main/java/com/powsybl/matpower/converter/MatpowerImporter.java

```diff
@@ -0,0 +1,469 @@
+            this.baseMva = baseMva;
+            this.ignoreBaseMva = ignoreBaseMva;
+        }
+
+        public boolean isIgnoreBaseMva() {
+            return ignoreBaseMva;
+        }
+
+        private double getBaseMva() {
+            return baseMva;
+        }
+    }
+
+    private static ContainersMapping createContainerMapping(MatpowerModel matpowerModel) {
+        ContainersMapping containersMapping = new ContainersMapping();
+
+        // group buses connected to non impedant lines to voltage levels
+        createVoltageLevelMapping(matpowerModel, containersMapping);
+
+        // group voltage levels connected by transformers to substations
+        createSubstationMapping(matpowerModel, containersMapping);
+
+        return containersMapping;
+    }
+
+    private static boolean isTransformer(MBranch branch) {
+        return branch.getRatio() != 0;
+    }
+
+    private static void createSubstationMapping(MatpowerModel model, ContainersMapping containersMapping) {
+        UndirectedGraph<String, Object> sGraph = new Pseudograph<>(Object.class);
+        for (String voltageLevelId : containersMapping.voltageLevelIdToBusNums.keySet()) {
+            sGraph.addVertex(voltageLevelId);
+        }
+        for (MBranch branch : model.getBranches()) {
+            if (isTransformer(branch)) {
+                sGraph.addEdge(containersMapping.busNumToVoltageLevelId.get(branch.getFrom()),
+                        containersMapping.busNumToVoltageLevelId.get(branch.getTo()));
+            }
+        }
+        int substationNum = 1;
+        for (Set<String> voltageLevelIds : new ConnectivityInspector<>(sGraph).connectedSets()) {
+            String substationId = "S" + substationNum++;
+            for (String voltageLevelId : voltageLevelIds) {
+                containersMapping.voltageLevelIdToSubstationId.put(voltageLevelId, substationId);
+            }
+        }
+    }
+
+    private static void createVoltageLevelMapping(MatpowerModel model, ContainersMapping containersMapping) {
+        UndirectedGraph<Integer, Object> vlGraph = new Pseudograph<>(Object.class);
+        for (MBus mBus : model.getBuses()) {
+            vlGraph.addVertex(mBus.getNumber());
+        }
+        for (MBranch mBranch : model.getBranches()) {
+            if (mBranch.getR() == 0 && mBranch.getX() == 0) {
+                vlGraph.addEdge(mBranch.getFrom(), mBranch.getTo());
+            }
+        }
+        for (Set<Integer> busNums : new ConnectivityInspector<>(vlGraph).connectedSets()) {
+            String voltageLevelId = "VL" + busNums.iterator().next();
+            containersMapping.voltageLevelIdToBusNums.put(voltageLevelId, busNums);
+            for (int busNum : busNums) {
+                containersMapping.busNumToVoltageLevelId.put(busNum, voltageLevelId);
+            }
+        }
+    }
+
+    private static String getBusId(int busNum) {
+        return "B" + busNum;
```

Comment:
Use `BUS` instead?

Reply:
my general remark for all the ID convention related feedbacks&suggestions: I tried to be as consistent as I could 
with respect to the existing cdf importer approach (ref. the iidm in the ieee test resources ) in order to have the same generated IDs
(and make it possible to directly compare the iidm, for example)

Reply:
fixed: used a prefix, consistently